R/xgboost-development.R

```diff
@@ -129,7 +129,8 @@ XGBoostDevelopment <- R6Class("XGBoostDevelopment",
                              data = self$xgb_trainMatrix,
                              nrounds = self$params$xgb_nrounds)
       # Save target list into the fit object for use in deploy
-      self$fitXGB$xgb_targetNames <- self$params$xgb_targetNames
+      # coerce the target names to characters to avoid factor subsetting issues
+      self$fitXGB$xgb_targetNames <- as.character(self$params$xgb_targetNames)
     },
```

Comment:
This change was made to avoid a subsetting issue.  The actual issue occurs in xgboost deployment (lines 259-261), but it seemed cleaner to make the change here.  Lines 259-261 of xgboost deployment are as follows:
```
# Set column names to match input targets
colnames(private$predictions)[1:self$params$xgb_numberOfClasses] <- self$params$xgb_targetNames
colnames(private$temp_predictions)[1:self$params$xgb_numberOfClasses] <- self$params$xgb_targetNames
```
To see what can go wrong, run the following code:
```
df <- data.frame(x1 = 1:5, x2 = sample(c('M','F'), size = 5, replace = T))
df

# Works as expected
new_cols <- as.factor(c("id", "gender"))
colnames(df) <- new_cols
df

# Assigns number corresponding to factor instead
colnames(df)[1:2] <- new_cols
df
```